File: Haskell/firstAssignment.hs
--Maurizio Altamura
--COP4020

multiplyTwo x y = x*y

multiplyThree x y z = x*y*z

first_a n = [x | x <- [1..n], (x `mod` 6 == 0) || (x `mod` 11 == 0)]

isMult6Or11 x = (x `mod` 6 == 0) || (x `mod` 11 == 0)
first_b n = [x | x <- [1..n], isMult6Or11 x]

second_a n = [x | x <- [1..n], (reverse(show x) == show x), head(show x) == '3']

isPalindromeThatStartsWithDigit3 n = (reverse(show n) == show n) && head(show n) == '3'
second_b n = [x | x <- [1..n], isPalindromeThatStartsWithDigit3 x]

File: Python/firstAssignment.py
#Maurizio Altamura

#problem 1
def problem1(x,y):
        print(x*y)

#problem 2
def problem2(x,y,z):
        return(x*y*z)

#problem 3
def first_a(n):
        return [x for x in range(n)
            if x % 6 == 0 or x % 11 == 0]
        
#problem 4
def first_b(n):
        def isMult6Or11(n):
            return( n % 6 == 0 or n % 11 ==0 )
        return [x for x in range (n)
            if isMult6Or11(x)]

#problem 5
def second_a(n):
        return [x for x in range (n)
            if str(x) == (str(x))[::-1] and str(x)[:1] == '3' ]

#problem 6
def second_b(n):
        def isPalindromeThatStartsWithDigit3(n):
            return n
        return [x for x in range (n)
            if str(x) == (str(x))[::-1] and str(x)[:1] == '3' ]
